sklearn/ensemble/tests/test_iforest.py

```diff
@@ -105,8 +106,17 @@ def test_iforest_error():
     assert_warns_message(UserWarning,
                          "max_samples will be set to n_samples for estimation",
                          IsolationForest(max_samples=1000).fit, X)
-    assert_no_warnings(IsolationForest(max_samples='auto').fit, X)
-    assert_no_warnings(IsolationForest(max_samples=np.int64(2)).fit, X)
+    with pytest.warns(None) as record:
+        IsolationForest(max_samples='auto').fit(X)
```

Comment:
Why does this result in a PendingDeprecationWarning as it stands?

Reply:
Why does anything result in a PendinDeprecationWarning? There are no matrices in sklearn except for a test in the input validation.